metis-media-service/src/main/java/eu/europeana/metis/mediaservice/MediaProcessor.java

```diff
@@ -0,0 +1,441 @@
+				double frameRate = Double.parseDouble(frameRateParts[0]) / Double.parseDouble(frameRateParts[1]);
+				resource.setFrameRete(frameRate);
+			} else if (audioStream != null) {
+				resource.setDuration(audioStream.getDouble("duration"));
+				resource.setBitrate(audioStream.getInt("bit_rate"));
+				resource.setCahhnels(audioStream.getInt("channels"));
+				resource.setSampleRate(audioStream.getInt("sample_rate"));
+				resource.setSampleSize(audioStream.getInt("bits_per_sample"));
+			} else {
+				throw new Exception("No media streams");
+			}
+		} catch (MediaException e) {
+			throw e;
+		} catch (Exception e) {
+			log.info("Could not parse ffprobe response:\n" + StringUtils.join(resultLines, "\n"), e);
+			throw new MediaException("File seems to be corrupted", "AUDIOVIDEO ERROR");
+		}
+	}
+	
+	private void processText(String url, String mimeType, File contents) throws IOException {
+		if (!shouldExtractMetadata(urlTypes.get(url)))
+			return;
+		
+		boolean containsText = mimeType.startsWith("text/");
+		Integer resolution = null;
+		
+		if (mimeType.equals("application/pdf")) {
+			try (PDDocument document = PDDocument.load(contents)) {
+				PDFTextStripper pdfStripper = new PDFTextStripper();
+				String text = pdfStripper.getText(document).replaceAll("\\s", "");
+				containsText = !text.isEmpty();
+				
+				firstImageSearch:
+				for (int i = 0; i < document.getNumberOfPages(); i++) {
+					PDResources res = document.getPage(i).getResources();
+					Iterator<COSName> it = res.getXObjectNames().iterator();
+					while (it.hasNext()) {
+						PDXObject xObject = res.getXObject(it.next());
+						if (!(xObject instanceof PDImageXObject))
+							continue;
+						PDMetadata pdMetadata = ((PDImageXObject) xObject).getMetadata();
+						if (pdMetadata == null)
+							break firstImageSearch;
+						String metadata = IOUtils.toString(pdMetadata.createInputStream(), "UTF-8");
+						Matcher resolutionMatcher = Pattern
+								.compile("<exif:XResolution>\\s*([0-9]+)\\s*</exif:XResolution>").matcher(metadata);
+						if (resolutionMatcher.find())
+							resolution = Integer.valueOf(resolutionMatcher.group(1));
+						break firstImageSearch;
+					}
+				}
+			}
+		}
+		
+		WebResource resource = edm.getWebResource(url);
+		resource.setMimeType(mimeType);
+		resource.setFileSize(contents.length());
+		resource.setContainsText(containsText);
+		resource.setResolution(resolution);
+	}
+	
+	private List<String> runCommand(List<String> command, boolean mergeError) throws IOException {
+		return runCommand(command, mergeError, null);
+	}
+	
+	private List<String> runCommand(List<String> command, boolean mergeError, byte[] inputBytes) throws IOException {
+		Process process = new ProcessBuilder(command).redirectErrorStream(mergeError).start();
+		if (!mergeError) {
+			commandIOThreadPool.execute(() -> {
+				try (InputStream errorStream = process.getErrorStream()) {
+					String output = IOUtils.toString(errorStream);
```

Comment:
Deprecated toString method. Please use the toString that has as second parameter the string encoding.
e.g. IOUtils.toString(errorStream, "UTF-8");